src/receiver.ts

```diff
@@ -189,213 +248,381 @@ export class Receiver {
     })
   }
 
+  protected async callRequestHandler (requestId: number, amount: string, data: Buffer, keyId?: Buffer): Promise<{ fulfill: boolean, responseData: Buffer }> {
+    let fulfill = false
+    let responseData = Buffer.alloc(0)
+
+    // This promise resolves when the user has either accepted or rejected the payment
+    await new Promise(async (resolve, reject) => {
+      // Reject the payment if:
+      // a) the user explicity calls reject
+      // b) if they don't call accept
+      // c) if there is an error thrown in the request handler
+      try {
+        await Promise.resolve(this.requestHandler({
+          keyId,
+          amount: new BigNumber(amount),
+          data,
+          accept: (userResponse = Buffer.alloc(0)) => {
+            fulfill = true
+            responseData = userResponse
+            resolve()
+          },
+          reject: (userResponse = Buffer.alloc(0)) => {
```

Comment:
given that these functions are synchronous, I believe you could call `reject` followed by `accept`, and the first call would be ignored. If you call `accept` followed by `reject`, it would fulfill the payment but use the responseData of the `reject` call. Either the responseData and fulfillment should be passed through the `resolve` call, or there should be a flag that makes sure you only call one of these functions.

Reply:
Added a check for this